src/lib/settings/settings.py

```diff
@@ -280,6 +280,32 @@ def get_metrics_collection_interval_min(self) -> int:
         return self.processed_settings[SettingFileNames.GENERAL]["tooling_environment"][
             "metrics_collection_interval_min"
         ]
+    
+    def get_grafana_mandatory_settings(self) -> tuple[str, str]:
+        """
+        Get mandatory grafana settings:
+            - grafana_vpc_id: VPC ID for Grafana instance
+            - grafana_security_group_id: Security Group ID for Grafana instance
+        If not provided, the Grafana stack will be skipped.
+        """
+        grafana_settings = self.processed_settings[SettingFileNames.GENERAL]["tooling_environment"]
+        return (
+            grafana_settings.get("grafana_vpc_id"),
+            grafana_settings.get("grafana_security_group_id"))
+
+    def get_grafana_optional_settings(self) -> tuple[str, str, str]:
+        """
+        Get optional grafana settings:
+            - grafana_key_pair_name: Key pair name. if not provided, will be created in the Grafana stack
+            - grafana_image: Bitnami Grafana image. Defaults to "bitnami-grafana-10.2.2-1-r02-linux-debian-11-x86_64-hvm-ebs-nami"
+            - grafana_instance_type: Grafana Instance type. Defaults to "t3.micro"
+        """
+        grafana_settings = self.processed_settings[SettingFileNames.GENERAL]["tooling_environment"]
```

Comment:
filling in default values, preferably, should be done when preparing processing settings (raw->processing)